src/securejoin.rs

```diff
@@ -606,6 +606,11 @@ pub(crate) async fn observe_securejoin_on_other_device(
                         return Ok(HandshakeMessage::Ignore);
                     }
                 };
+                if !peerstate.is_using_verified_key() {
+                    // The `if` is purely for performance:
+                    // If the verified key was used already, then the 1:1 chat was already protected.
+                    ChatId::set_protection_for_contact(context, contact_id).await?;
+                }
```

Comment:
This change is not related to https://github.com/deltachat/securejoin/blob/main/uxscenarios/resetup.md#i-happy-path-and-resetup-scenarios-with-green-checkmarked-chatting, but another problem I noticed